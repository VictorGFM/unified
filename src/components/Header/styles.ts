import styled from 'styled-components'
import { COLORS } from '../../styles/global'

export const HeaderDiv = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;

  width: 100%;
  height: 70px;
  border-bottom: 1px solid ${COLORS.GREY.THREE};

  background-color: ${COLORS.WHITE.ONE};
`

export const SpaceDiv = styled.div`
  flex: 1;
`

export const Logo = styled.img`
  flex: 1;

  width: 138px;
  height: 35px;

  cursor: pointer;
`

export const ButtonsDiv = styled.div`
  display: flex;
  flex: 1;
  justify-content: flex-end;
  gap: 20px;

  width: 376px;
  height: 46px;
  margin-right: 20px;
`
