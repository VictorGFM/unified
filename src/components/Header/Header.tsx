import { HeaderDiv, SpaceDiv, Logo, ButtonsDiv } from './styles'
import {
  AppLogo,
  HomeIcon,
  GroupsIcon,
  ContactsIcon,
  MessagesIcon,
  NotificationsIcon,
  ProfilePhotoIcon,
} from '../../assets'
import HeaderButton from '../HeaderButton/HeaderButton'

const Header = () => {
  return (
    <>
      <HeaderDiv>
        <SpaceDiv />
        <Logo src={AppLogo} />
        <ButtonsDiv>
          <HeaderButton icon={HomeIcon} />
          <HeaderButton icon={GroupsIcon} />
          <HeaderButton icon={ContactsIcon} />
          <HeaderButton icon={MessagesIcon} />
          <HeaderButton icon={NotificationsIcon} />
          <HeaderButton icon={ProfilePhotoIcon} />
        </ButtonsDiv>
      </HeaderDiv>
    </>
  )
}

export default Header
