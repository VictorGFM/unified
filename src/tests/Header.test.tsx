import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
import { describe, expect, it } from 'vitest'
import { render } from '@testing-library/react'
import Header from '../components/Header'

describe('<Header/>', () => {
  it('should show match snapshot', () => {
    const component = render(
      <Router>
        <Routes>
          <Route path="*" element={<Header />} />
        </Routes>
      </Router>
    )

    expect(component).toMatchSnapshot()
  })
})
