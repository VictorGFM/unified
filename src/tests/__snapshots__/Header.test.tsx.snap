// Vitest Snapshot v1

exports[`<Header/> > should show match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ kPWdKs"
      >
        <img
          class="sc-gsnTZi fcoXBT"
          src="/src/assets/icons/AppLogo.svg"
        />
        <div
          class="sc-dkzDqf hCLNHX"
        >
          <div
            class="sc-hKMtZM YDUQe"
            data-cy="header-button-home"
          >
            <div
              class="sc-eCYdqJ ipwMuM"
            >
              <img
                class="sc-jSMfEi jGzQWq"
                src="/src/assets/icons/HomeIcon.svg"
              />
            </div>
          </div>
          <div
            class="sc-hKMtZM YDUQe"
            data-cy="header-button-groups"
          >
            <div
              class="sc-eCYdqJ ipwMuM"
            >
              <img
                class="sc-jSMfEi jGzQWq"
                src="/src/assets/icons/GroupsIcon.svg"
              />
            </div>
          </div>
          <div
            class="sc-hKMtZM YDUQe"
            data-cy="header-button-contacts"
          >
            <div
              class="sc-eCYdqJ ipwMuM"
            >
              <img
                class="sc-jSMfEi jGzQWq"
                src="/src/assets/icons/ContactsIcon.svg"
              />
            </div>
          </div>
          <div
            class="sc-hKMtZM YDUQe"
            data-cy="header-button-messages"
          >
            <div
              class="sc-eCYdqJ ipwMuM"
            >
              <img
                class="sc-jSMfEi jGzQWq"
                src="/src/assets/icons/MessagesIcon.svg"
              />
            </div>
            <div
              class="sc-gKXOVf eSfgBS"
            >
              +30
            </div>
          </div>
          <div
            class="sc-hKMtZM YDUQe"
            data-cy="header-button-notifications"
          >
            <div
              class="sc-eCYdqJ ipwMuM"
            >
              <img
                class="sc-jSMfEi jGzQWq"
                src="/src/assets/icons/NotificationsIcon.svg"
              />
            </div>
            <div
              class="sc-gKXOVf eSfgBS"
            >
              +30
            </div>
          </div>
          <div
            class="sc-hKMtZM YDUQe"
          >
            <div
              class="sc-eCYdqJ BvGKt"
            >
              <img
                class="sc-jSMfEi bsZMGe"
                src="/src/assets/icons/ProfilePhotoIcon.svg"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ kPWdKs"
    >
      <img
        class="sc-gsnTZi fcoXBT"
        src="/src/assets/icons/AppLogo.svg"
      />
      <div
        class="sc-dkzDqf hCLNHX"
      >
        <div
          class="sc-hKMtZM YDUQe"
          data-cy="header-button-home"
        >
          <div
            class="sc-eCYdqJ ipwMuM"
          >
            <img
              class="sc-jSMfEi jGzQWq"
              src="/src/assets/icons/HomeIcon.svg"
            />
          </div>
        </div>
        <div
          class="sc-hKMtZM YDUQe"
          data-cy="header-button-groups"
        >
          <div
            class="sc-eCYdqJ ipwMuM"
          >
            <img
              class="sc-jSMfEi jGzQWq"
              src="/src/assets/icons/GroupsIcon.svg"
            />
          </div>
        </div>
        <div
          class="sc-hKMtZM YDUQe"
          data-cy="header-button-contacts"
        >
          <div
            class="sc-eCYdqJ ipwMuM"
          >
            <img
              class="sc-jSMfEi jGzQWq"
              src="/src/assets/icons/ContactsIcon.svg"
            />
          </div>
        </div>
        <div
          class="sc-hKMtZM YDUQe"
          data-cy="header-button-messages"
        >
          <div
            class="sc-eCYdqJ ipwMuM"
          >
            <img
              class="sc-jSMfEi jGzQWq"
              src="/src/assets/icons/MessagesIcon.svg"
            />
          </div>
          <div
            class="sc-gKXOVf eSfgBS"
          >
            +30
          </div>
        </div>
        <div
          class="sc-hKMtZM YDUQe"
          data-cy="header-button-notifications"
        >
          <div
            class="sc-eCYdqJ ipwMuM"
          >
            <img
              class="sc-jSMfEi jGzQWq"
              src="/src/assets/icons/NotificationsIcon.svg"
            />
          </div>
          <div
            class="sc-gKXOVf eSfgBS"
          >
            +30
          </div>
        </div>
        <div
          class="sc-hKMtZM YDUQe"
        >
          <div
            class="sc-eCYdqJ BvGKt"
          >
            <img
              class="sc-jSMfEi bsZMGe"
              src="/src/assets/icons/ProfilePhotoIcon.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
